presto-common/src/main/java/com/facebook/presto/common/type/TypeUtils.java

```diff
@@ -101,4 +111,28 @@ static void checkElementNotNull(boolean isNull, String errorMsg)
             throw new NotSupportedException(errorMsg);
         }
     }
+
+    static <V> Map<String, V> normalizeEnumEntries(Optional<String> enumName, Map<String, V> entries)
+    {
+        enumName.ifPresent(name -> requireNonNull(name, "Enum name cannot be null"));
```

Comment:
I will actually remove this logic with the optional name altogether. this was mainly to help users with debugging but there will have to be code elsewhere to catch those errors anyway, so we can log the name of the offending enum at that point instead of here.